mslice/models/workspacemanager/mantid_workspace_provider.py

```diff
@@ -132,7 +132,10 @@ def get_emode(self, workspace):
             workspace_handle = self.get_workspace_handle(workspace)
         else:
             workspace_handle = workspace
-        emode = workspace_handle.getEMode().name
+        try:
+            emode = workspace_handle.getEMode().name
+        except AttributeError:
+            return None
```

Comment:
I'm not sure about silently swallowing this. If we can't retrieve it then the caller needs to know. How about we model it on Python's [`dict.get()` method](https://docs.python.org/2/library/stdtypes.html#dict.get)? The caller can choose whether to see an exception or not and then handle it appropriately.